eZ/Publish/Core/IO/ConfigScopeChangeAwareIOService.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\IO;
+
+use eZ\Publish\Core\IO\Values\BinaryFile;
+use eZ\Publish\Core\IO\Values\BinaryFileCreateStruct;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\SiteAccess;
+use eZ\Publish\SPI\MVC\EventSubscriber\ConfigScopeChangeSubscriber;
+
+class ConfigScopeChangeAwareIOService implements IOServiceInterface, ConfigScopeChangeSubscriber
+{
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    protected $configResolver;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface */
+    protected $innerIOService;
+
+    /** @var string */
+    protected $prefixParameterName;
```

Comment:
Protected? As mentioned in previous command. Please check all that cases, now I will stop with comments about it

Reply:
Why do you want everything to be private? It very much kills the possibility of extending the class. There is no BC guarantee on `Core` but we might extend it in our product.